salt/master.py

```diff
@@ -1269,9 +1269,12 @@ def _file_recv(self, load):
             return {}
         load.pop('tok')
 
+        # Join path
+        sep_path = os.sep.join(load['path'])
```

Comment:
Ah, never mind. Ignore the point about the tuple -- I thought you were expecting a behavior difference but I misunderstood. I see the difference here. There's a behavior difference between `os.sep.join` (which requires an iterable) and `os.path.join` which takes (`path`, `*path`). This approach is fine.